### PR TITLE
Update `enableEvent()`

### PR DESCRIPTION
**Pull request by Lukellmann**
Updates the extension function `enableEvents()` for `Intents.IntentsBuilder` to support all new events that were added to Kord to match [this list](https://discord.com/developers/docs/topics/gateway#list-of-intents).

This also includes little fixes:
- the `DirectMessages` intent will no longer be added to events that only need the `Guilds` intent
- `TypingStartEvent` will now add both `GuildMessageTyping` and `DirectMessageTyping`

Other changes:
- add `LoginBuilder.intents()` convenience function

### CODE REVIEW

**Comment by Lukellmann** [comment]
While working on this I also noticed that some events are missing in Kord, namely:
```
STAGE_INSTANCE_CREATE
STAGE_INSTANCE_UPDATE
STAGE_INSTANCE_DELETE

GUILD_STICKERS_UPDATE

INTEGRATION_CREATE
INTEGRATION_UPDATE
INTEGRATION_DELETE
```
We can address this in a separate PR.

File: core/src/main/kotlin/Util.kt [hunk 1]
```diff
@@ -317,48 +331,105 @@ public fun Intents.IntentsBuilder.enableEvent(event: KClass<out Event>): Unit =
...
+    NewsChannelThreadCreateEvent::class,
+    TextChannelThreadCreateEvent::class,
+    UnknownChannelThreadCreateEvent::class,
+
+    ThreadUpdateEvent::class,
+    NewsChannelThreadUpdateEvent::class,
+    TextChannelThreadUpdateEvent::class,
+    UnknownChannelThreadUpdateEvent::class,
+
+    ThreadChannelDeleteEvent::class,
```
Comment (HopeBaron) [inline core/src/main/kotlin/Util.kt:370]: ThreadChannelDeleteEvent is enough here, the other two are sub types

File: core/src/main/kotlin/Util.kt [hunk 2]
```diff
@@ -317,48 +331,105 @@ public fun Intents.IntentsBuilder.enableEvent(event: KClass<out Event>): Unit =
...
     TextChannelDeleteEvent::class,
+    UnknownChannelDeleteEvent::class,
     VoiceChannelDeleteEvent::class,
 
-    ChannelPinsUpdateEvent::class,
-    -> {
-        +Guilds
-        +DirectMessages
-    }
+    ThreadChannelCreateEvent::class,
```
Comment (HopeBaron) [inline core/src/main/kotlin/Util.kt:360]: Same here.

File: core/src/main/kotlin/Util.kt [hunk 3]
```diff
@@ -317,48 +331,105 @@ public fun Intents.IntentsBuilder.enableEvent(event: KClass<out Event>): Unit =
...
+    NewsChannelThreadCreateEvent::class,
+    TextChannelThreadCreateEvent::class,
+    UnknownChannelThreadCreateEvent::class,
+
+    ThreadUpdateEvent::class,
```
Comment (HopeBaron) [inline core/src/main/kotlin/Util.kt:365]: Same here

**Review by HopeBaron — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/Util.kt (hunk 1 above)
Comment (Lukellmann) [reply core/src/main/kotlin/Util.kt:370]: This does not work, `when` uses `equals`, from `KClass.equals`:
> Returns true if this KClass instance represents the same Kotlin class as the class represented by other. On JVM this means that all of the following conditions are satisfied:
> 1. other has the same (fully qualified) Kotlin class name as this instance.
> 2. other's backing Class object is loaded with the same class loader as the Class object of this instance.
>
> [...]

Also see this Kotlin Playground: https://pl.kotl.in/H1e13i-bw

**Pull request merged**